db/disk/course.go

```diff
@@ -124,6 +127,11 @@ func (this *backend) GetCourses() (map[string]*model.Course, error) {
             return nil, fmt.Errorf("Failed to load course '%s': '%w'.", configPath, err);
         }
 
+        func() {
+            common.ReadLock(course.GetID());
+            defer common.ReadUnlock(course.GetID());
```

Comment:
Also, there is no logging on the error (mentioned via email).

Reply:
It should NOT be done async.
Everything that you need to be inside the function should be in it.
(Sorry for the somewhat recursive definition, but you just need to put whatever you need to in it.)